openchemistry/__init__.py

```diff
@@ -348,17 +210,25 @@ def __init__(self, _id, cjson):
         super(GirderMolecule, self).__init__(CjsonProvider(cjson))
         self._id = _id
 
-    def optimize(self, basis=None, theory=None, functional=None, code='nwchem'):
-        return _optimize(self._id, basis, theory, functional, code=code)
+    def calculate(self, container_name, input_parameters, input_geometry=None):
```

Comment:
@cryos How do you feel about this name?

Reply:
I was just starting to look at this, I don't think we should get too attached to it. I also think that even if we have an underlying generic call to execute_code or similar, that we will still want to have some high level calls that might just add type = optimize under the covers and dispatch to a very generic function.

This is part of what we were wanting to rework some, and I think it would benefit from some discussion.